service/api_task.go

```diff
@@ -194,6 +194,19 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if projectRef.CommitQueue.Enabled && details.Status != evergreen.TaskSucceeded {
+		env := evergreen.GetEnvironment()
+		githubStatusSender, err := env.GetSender(evergreen.SenderGithubStatus)
```

Comment:
I would move this into TryDequeueAndAbortCommitQueueVersion since there's no need to get a sender in the API here